File: Tutorial2-2SelectContructorAndFieldInjection/src/main/java/com/example/tutorial2selectcontructorandfieldinjection/di/MyComponent.java
package com.example.tutorial2selectcontructorandfieldinjection.di;


import com.example.tutorial2selectcontructorandfieldinjection.MainActivity;
import javax.inject.Singleton;
import dagger.Component;

@Component(modules = SharedPreferencesModule.class)
@Singleton
public interface MyComponent {
    void inject(MainActivity mainActivity);
}

File: Tutorial2-2SelectContructorAndFieldInjection/src/main/java/com/example/tutorial2selectcontructorandfieldinjection/di/MyComponent.kt
package com.example.tutorial2selectcontructorandfieldinjection.di

import com.example.tutorial2selectcontructorandfieldinjection.MainActivity
import dagger.Component
import javax.inject.Singleton

@Component(modules = [SharedPreferencesModule::class])
@Singleton
interface MyComponent {
    fun inject(mainActivity: MainActivity)
}